### PR TITLE
Migrations to rebuild main view indexes with id included in ordering

### PR DESCRIPTION
**Pull request by rowanseymour**

### CODE REVIEW

File: temba/msgs/migrations/0058_indexes_update.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+-- for the Incoming folder (API only)
+CREATE INDEX msgs_msg_incoming_folder
+ON msgs_msg (org_id, modified_on DESC, id DESC)
+WHERE direction = 'I';
+
+DROP INDEX IF EXISTS msg_direction_modified_inbound;
+
+-- for the Outbox folder
+CREATE INDEX msgs_msg_outbox_folder
```
Comment (rowanseymour) [inline temba/msgs/migrations/0058_indexes_update.py]: This is what meant about naming indexes - this index is for a specific view so it's simpler if that's reflected in the index name rather than trying to encode the entire index criteria into the name

Comment (nicpottier) [inline temba/msgs/migrations/0058_indexes_update.py]: So personally I don't like this, because the naming is making promises it can't keep. We don't know that that index is only used by that view and we don't know even if it did that it would continue to in the future. Postgres picks up crazy indexes all over the place, so I think us trying to code in the "use" of an index is going to drift from reality really quickly even if we get it right at the start.

It also begs the question of how we would name indexes that get used by more than one view.

In short, I prefer the current naming but willing to be outvoted. @ericnewcomer @norkans7

Comment (norkans7) [inline temba/msgs/migrations/0058_indexes_update.py]: Yes, index can be used in many places. we can keep using the TABLE_NAME_FIELDS_NAME_WHERE_FIELDS as flows_flowrun_org_id_modified_on_id_responded

Sometimes we change view name too

Comment (rowanseymour) [inline temba/msgs/migrations/0058_indexes_update.py]: For generic indexes sure. But an index like...

```sql
CREATE INDEX msgs_msg_sent_folder
ON msgs_msg(org_id, created_on DESC, id DESC)
WHERE direction = 'O' AND visibility = 'V' AND status IN ('W', 'S', 'D');
```

exists to serve the sent folder view. If the Sent folder view changes - we need to change this index. If it gets used somewhere else then great. What would we call it that describes all its criteria?

Comment (nicpottier) [inline temba/msgs/migrations/0058_indexes_update.py]: They do get weird when there are conditions... maybe a standard like: `msgs_msg_org_created_id_where_outbound_visible_sent`

Comment (rowanseymour) [inline temba/msgs/migrations/0058_indexes_update.py]: Meh I still think it's more useful to see from an index name if it has a specific purpose.. but whatever the consensus is @ericnewcomer

Comment (ericnewcomer) [inline temba/msgs/migrations/0058_indexes_update.py]: I prefer a name that describes the nature of what the index is on not what it is for. While you might build it for a specific purpose, other uses of that index will likely occur. It could be misleading to suggest otherwise by naming it for its original use.

Comment (rowanseymour) [inline temba/msgs/migrations/0058_indexes_update.py]: IMO if another bit of code starts using the sent view index (which is very specific) then we're almost certainly doing something wrong, would like to know about it, and probably missing a more tailored index in that place. But `msgs_msg_org_created_id_where_outbound_visible_sent` it is...

**Comment by rowanseymour** [comment]
Database index names updated according to our new documented convention: https://github.com/nyaruka/nyaruka/wiki/Database-Index-Naming-Convention

**Comment by nicpottier** [comment]
Occurred to me we probably need to resize the db before shipping this.. :(
